src/components/Editor.jsx

```diff
@@ -136,12 +136,60 @@ class Editor extends React.Component {
     session.on('change', () => {
       this.props.onInput(this._editor.getValue());
     });
+    session.selection.on('changeCursor', () => {
+      let selector = '';
+      if (this.props.language === 'css') {
+        selector = this._locateSelectorCSS(session);
+      }
+      this._sendSelectorToHighlighter(selector);
+    });
     session.setAnnotations(this.props.errors);
     this._editor.setSession(session);
     this._editor.moveCursorTo(0, 0);
     this._resizeEditor();
   }
 
+  _locateSelectorCSS(session) {
+    const reCssQuery = /(^|.*\})(.*)\{|\}/;
```

Comment:
So my high level feedback on this is that regular expressions don’t make very good language parsers. This is both because regexp-based parsing tends to be hard to understand, and also because it tends to be brittle and miss edge cases. I can tell you that at least the first concern applies here ; )

My pitch would be to instead use a bona fide CSS parser to do the job. I looked at a few options and I think [`postcss`](https://github.com/postcss/postcss) is the move. `postcss` is already a dependency of `stylelint`, so using it here won’t mean incorporating another library and increasing bundle size (although I do think we should load it asynchronously with `System.import()`).

I think the task at hand is fairly simple with `postcss`, to wit:

* Iterate over every rule in the parsed stylesheet using [`root.walkRules()`](http://api.postcss.org/Root.html#walkRules)
* Check to see if the cursor position falls within the rule boundaries specified in the `source` property of the rule. If not, move on.
* Calculate the bounds of the rule’s selector. The beginning of the selector is imply the beginning of the rule. You can calculate the end of the selector by getting the length of the `selector` property of the rule, and then adding it to the beginning. (Note that properly handling multi-line selectors would be a mildly more complicated version of the same process)
* See if the cursor position falls within the selector bounds you calculated

Let me know if this sounds reasonable! I think ultimately it’ll be no more code than what you’ve got here, but probably considerably easier to take in and more robust.

Reply:
Yes - this sounds good! Much simpler and easier to understand. Will try it out! Do you think it makes sense to do only if the cursor is within the bounds selector text or the entire rule?

Reply:
I don’t feel super-strongly about it but intuitively I think just the selector itself would feel the most natural.

Also, I might have forgotten to mention this, but I think we should only show the highlights when the editor is focused (I think that would also reduce the pressure on the frame-side code to keep the highlight boxes perfectly in sync at all times)